Valeriia/Movies_database_project/src/components/SearchForm/SearchForm.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { Component } from "react";
+import { withRouter } from "react-router-dom";
+import { RouteComponentProps } from "react-router";
+import { parse } from "query-string";
+import Input from "../Input";
+import Button from "../Button";
+import "./SearchForm.scss";
+
+interface ISearchFormProps extends RouteComponentProps {
+  onSearchClick(searchTerm: string, filterBy: string): void;
+}
+
+interface ISearchFormState {
+  searchTerm: string;
+  filterBy: string;
+}
+
+class SearchForm extends Component<ISearchFormProps, ISearchFormState> {
+  state = {
+    searchTerm: "",
+    filterBy: "title",
+  };
+
+  componentDidMount = () => {
+    const query = parse(this.props.location.search) as {
+      filterBy: string;
+    };
+    const { filterBy } = query;
+    const filterByType = filterBy ? filterBy : "title";
+    this.setState({
+      filterBy: filterByType,
+    });
+  };
+
+  componentDidUpdate = (prevProps: ISearchFormProps) => {
+    if (this.props.location.search !== prevProps.location.search) {
```

Comment:
I would add one more condition 
`this.props.location.search !== prevProps.location.search && this.props.history.action === "POP" `

Because now every time when user submits new value, you make history push and then componentDidMount get trigered and you update state, that don't need to be updated